### PR TITLE
Add scopes for on-demand=true client

### PR DESCRIPTION
**Pull request by zhichengliu12581**
Add scopes for on-demand=true client and login with azure client

### CODE REVIEW

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapi/AzureActiveDirectoryResourceServerClientConfiguration.java [hunk 1]
```diff
@@ -40,7 +40,8 @@ public AzureClientRegistrationRepository azureClientRegistrationRepository() {
         return new AzureClientRegistrationRepository(
             clientRegistrationInitialization.createDefaultClient(),
             clientRegistrationInitialization.createAuthzClients(),
-            aadAuthenticationProperties);
+            aadAuthenticationProperties,
+            false);
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapi/AzureActiveDirectoryResourceServerClientConfiguration.java:44]: Delete booleam parameter, use different class for webapp and webapi.
```
AADClientRegistrationRepository
AADWebAppClientRegistrationRepository
AADWebApiClientRegistrationRepository
```

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapp/AzureActiveDirectoryConfiguration.java [hunk 2]
```diff
@@ -126,6 +127,13 @@ private void addAzureConfiguredScopes(Set<String> result) {
 
     private ClientRegistration createClientBuilder(String id, AuthorizationProperties authz) {
         ClientRegistration.Builder result = createClientBuilder(id);
+        List<String> scopes = authz.getScopes();
+        if (authz.isOnDemand() && !scopes.contains("openid")) {
+            scopes.add("openid");
+        }
+        if (authz.isOnDemand() && !scopes.contains("profile")) {
+            scopes.add("profile");
+        }
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapp/AzureActiveDirectoryConfiguration.java:136]:
```
if (authz.isOnDemand()) {
    if()...
    if()...
}
```

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapp/AzureActiveDirectoryConfiguration.java [hunk 3]
```diff
@@ -126,6 +127,13 @@ private void addAzureConfiguredScopes(Set<String> result) {
...
         result.scope(authz.getScopes());
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapp/AzureActiveDirectoryConfiguration.java:137]: result.scope(scopes);

File: sdk/spring/azure-spring-boot/src/test/java/com/azure/spring/aad/webapp/AzureActiveDirectoryConfigurationTest.java [hunk 4]
```diff
@@ -144,9 +144,8 @@ public void aadAwareClientRepository() {
                 assertTrue(clientRepo.isAuthzClient("graph"));
 
                 List<ClientRegistration> clients = collectClients(clientRepo);
-                assertEquals(2, clients.size());
-                assertEquals("graph", clients.get(0).getRegistrationId());
-                assertEquals("azure", clients.get(1).getRegistrationId());
+                assertEquals(1, clients.size());
+                assertEquals("azure", clients.get(0).getRegistrationId());
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot/src/test/java/com/azure/spring/aad/webapp/AzureActiveDirectoryConfigurationTest.java:148]: Add test for both configurations.

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapp/AzureActiveDirectoryConfiguration.java [hunk 5]
```diff
@@ -51,7 +51,8 @@ public AzureClientRegistrationRepository clientRegistrationRepository() {
         return new AzureClientRegistrationRepository(
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapp/AzureActiveDirectoryConfiguration.java:51]: Rename this file to `AADWebAppConfiguration`

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapi/AzureActiveDirectoryResourceServerClientConfiguration.java [hunk 6]
```diff
@@ -40,7 +40,8 @@ public AzureClientRegistrationRepository azureClientRegistrationRepository() {
         return new AzureClientRegistrationRepository(
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapi/AzureActiveDirectoryResourceServerClientConfiguration.java:40]: Rename this file to `AADWebApiConfiguration`

**Review by chenrujun — CHANGES_REQUESTED** [review]

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapp/AADWebAppConfiguration.java [hunk 7]
```diff
@@ -161,6 +161,15 @@ private void addAzureConfiguredScopes(Set<String> result) {
...
         ClientRegistration.Builder result = createClientBuilder(id);
+        List<String> scopes = authz.getScopes();
+        if (authz.isOnDemand()) {
+            if (!scopes.contains("openid")) {
+                scopes.add("openid");
+            }
+            if (!scopes.contains("profile")) {
+                scopes.add("profile");
+            }
+        }
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapp/AADWebAppConfiguration.java:172]: This is not updated yet.

Comment (lzc-1997-abel) [reply sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapp/AADWebAppConfiguration.java:172]: sorry , i missed that

Comment (lzc-1997-abel) [reply sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapp/AADWebAppConfiguration.java:172]: done

**Comment by chenrujun** [comment]
Hi, @lzc-1997-abel , please resolve the conflicts.

**Review by chenrujun — APPROVED** [review]

**Pull request merged**